src/components/Common/UI/ProgressBreadcrumbs/ProgressBreadcrumbsTypes.ts

```diff
@@ -0,0 +1,62 @@
+import type { OnEventType } from '@/components/Base/useBase'
+import type { EventType } from '@/types/Helpers'
+
+export interface Breadcrumb {
+  /**
+   * Unique key for the breadcrumb step
+   */
+  key: string
+  /**
+   * Translation key for the breadcrumb label
+   */
+  label: string
+  /**
+   * Optional translation namespace
+   */
+  namespace?: string
+  /**
+   * Optional variables for the breadcrumb label
+   */
+  variables?: Record<string, unknown>
+  /**
+   * Event handler for breadcrumb navigation
+   */
+  onNavigate?: (context: unknown) => unknown
+}
+
+export interface BreadcrumbNode {
+  /**
+   * Parent node key (null for root nodes)
+   */
+  parent: string | null
+  /**
+   * The breadcrumb item data
+   */
+  item: Breadcrumb
+}
+
+export type BreadcrumbNodes = Record<string, BreadcrumbNode>
+export type BreadcrumbTrail = Record<string, Breadcrumb[]>
```

Comment:
My hope would be that we can remove these to keep the interface that adapter consumers need to deal with as simple as possible.

If there is logic like this that helps us facilitate the Breadcrumb implementation, we can always create a codebase specific version that lives in our common components directory similar to what HamburgerMenu is doing for Menu. Basically we want the adapter to only have to expose the bare minimum for the consumer to wire up